pkg/federation-controller/service/dns/dns_test.go

```diff
@@ -290,17 +368,165 @@ func TestServiceController_ensureDnsRecords(t *testing.T) {
 			if err != nil {
 				t.Errorf("error querying rr for zone %q: %v", zoneName, err)
 			}
+			availableDNSRecords = sets.NewString()
 			for _, rr := range rrList {
 				rrdatas := rr.Rrdatas()
 
 				// Put in consistent (testable-by-string-comparison) order
 				sort.Strings(rrdatas)
-				records.Insert(fmt.Sprintf("%s:%s:%s:%d:%s", zoneName, rr.Name(), rr.Type(), rr.Ttl(), rrdatas))
+				availableDNSRecords.Insert(fmt.Sprintf("%s:%s:%s:%d:%s", zoneName, rr.Name(), rr.Type(), rr.Ttl(), rrdatas))
 			}
 		}
 
-		if !records.Equal(test.expected) {
-			t.Errorf("Test %q failed.  Actual=%v, Expected=%v", test.name, records, test.expected)
+		if !availableDNSRecords.Equal(expectedDNSRecords) {
+			return false, nil
 		}
+		return true, nil
+	})
+	if err != nil {
+		t.Errorf("Actual DNS records does not match expected. Actual=%v, Expected=%v", availableDNSRecords, expectedDNSRecords)
 	}
 }
+
+func TestNewServiceDNSController(t *testing.T) {
```

Comment:
[Clarification]: I did not clearly understand what additionally is this function testing, which is not already covered in TestServiceDNSController(). Is this mainly to improve code coverage?

Reply:
TestNewServiceDNSController is testing the initialisation parameters passed to the controller. The name is confusing, i will change it.